bgpd/bgp_mplsvpn.c

```diff
@@ -823,15 +823,15 @@ static struct bgp_path_info *
 leak_update(struct bgp *bgp, /* destination bgp instance */
 	    struct bgp_dest *bn, struct attr *new_attr, /* already interned */
 	    afi_t afi, safi_t safi, struct bgp_path_info *source_bpi,
-	    mpls_label_t *label, uint32_t num_labels, void *parent,
-	    struct bgp *bgp_orig, struct prefix *nexthop_orig,
-	    int nexthop_self_flag, int debug)
+	    mpls_label_t *label, uint32_t num_labels, struct bgp *bgp_orig,
+	    struct prefix *nexthop_orig, int nexthop_self_flag, int debug)
 {
 	const struct prefix *p = bgp_dest_get_prefix(bn);
 	struct bgp_path_info *bpi;
 	struct bgp_path_info *new;
 	struct bgp_path_info_extra *extra;
 	uint32_t num_sids = 0;
+	void *parent = (void *)source_bpi;
```

Comment:
Explicit typecasting is not needed here. 

```c
void *parent = source_bpi;
```

Reply:
fixed